### PR TITLE
Allow mask(jnp.split)

### PR DESCRIPTION
**Pull request by juliuskunze**
Checks that polymorphic shapes are propagated correctly through `jnp.split`, allowing it to be part of masked functions. This required `jnp.split` to calculate indices by itself, instead of relying on result shapes of `np.split` called with dummy numpy arrays. This should also make non-jitted `jnp.split` faster.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Excellent, thanks. It's satisfying to remove that `dummy_val` stuff too!
